lib/model/model.go

```diff
@@ -1142,6 +1152,52 @@ func (m *Model) handleDeintroductions(introducerCfg config.DeviceConfiguration,
 	return changed
 }
 
+// handleAutoAccepts handles adding and sharing folders for devices that have
+// AutoAcceptFolders set to true.
+func (m *Model) handleAutoAccepts(deviceCfg config.DeviceConfiguration, folder protocol.Folder) bool {
+	if _, ok := m.cfg.Folder(folder.ID); !ok {
+		defaultPath := m.cfg.Options().DefaultFolderPath
+		defaultPathFs := fs.NewFilesystem(fs.FilesystemTypeBasic, defaultPath)
+		for _, path := range []string{folder.Label, folder.ID} {
+			if _, err := defaultPathFs.Lstat(path); !fs.IsNotExist(err) {
+				continue
+			}
+
+			fcfg := config.NewFolderConfiguration(folder.ID, fs.FilesystemTypeBasic, filepath.Join(defaultPath, path))
+			fcfg.Label = folder.Label
+
+			// Need to wait for the promise, as this calls CommitConfiguration,
```

Comment:
Still "promise" plus all the `p` variables don't really make sense anymore with the waiter name change - see my more general comment about that: https://github.com/syncthing/syncthing/pull/4551#issuecomment-347137797